src/EditorFeatures/CSharpTest/UseCompoundAssignment/UseCompoundAssignmentTests.cs

```diff
@@ -0,0 +1,574 @@
+}",
+@"public class C
+{
+    int a;
+
+    void M()
+    {
+        this.a += 10;
+    }
+}");
+        }
+
+        [Fact, Trait(Traits.Feature, Traits.Features.CodeActionsUseCompoundAssignment)]
+        public async Task TestTriviaInsensitive()
+        {
+            await TestInRegularAndScriptAsync(
+@"public class C
+{
+    int a;
+
+    void M()
+    {
+        this  .  /*trivia*/ a [||]= this /*comment*/ .a + 10;
+    }
+}",
+@"public class C
+{
+    int a;
+
+    void M()
+    {
+        this  .  /*trivia*/ a += 10;
+    }
+}");
+        }
+
+        [Fact, Trait(Traits.Feature, Traits.Features.CodeActionsUseCompoundAssignment)]
+        public async Task TestStaticFieldThroughType()
+        {
+            await TestInRegularAndScriptAsync(
+@"public class C
+{
+    static int a;
+
+    void M()
+    {
+        C.a [||]= C.a + 10;
+    }
+}",
+@"public class C
+{
+    static int a;
+
+    void M()
+    {
+        C.a += 10;
+    }
+}");
+        }
+
+        [Fact, Trait(Traits.Feature, Traits.Features.CodeActionsUseCompoundAssignment)]
+        public async Task TestStaticFieldThroughNamespaceAndType()
+        {
+            await TestInRegularAndScriptAsync(
+@"namespace NS
+{
+    public class C
+    {
+        static int a;
+
+        void M()
+        {
+            NS.C.a [||]= NS.C.a + 10;
+        }
+    }
+}",
+@"namespace NS
+{
+    public class C
+    {
+        static int a;
+
+        void M()
+        {
+            NS.C.a += 10;
+        }
+    }
+}");
+        }
+
+        [Fact, Trait(Traits.Feature, Traits.Features.CodeActionsUseCompoundAssignment)]
+        public async Task TestParenthesized()
+        {
+            await TestInRegularAndScriptAsync(
+@"public class C
+{
+    int a;
+
+    void M()
+    {
+        (a) [||]= (a) + 10;
+    }
+}",
+@"public class C
+{
+    int a;
+
+    void M()
+    {
+        (a) += 10;
+    }
+}");
+        }
+
+        [Fact, Trait(Traits.Feature, Traits.Features.CodeActionsUseCompoundAssignment)]
+        public async Task TestThroughBase()
+        {
+            await TestInRegularAndScriptAsync(
+@"public class C
+{
+    public int a;
+}
+
+public class D : C
+{
+    void M()
+    {
+        base.a [||]= base.a + 10;
+    }
+}",
+@"public class C
+{
+    public int a;
+}
+
+public class D : C
+{
+    void M()
+    {
+        base.a += 10;
+    }
+}");
+        }
+
+        [Fact, Trait(Traits.Feature, Traits.Features.CodeActionsUseCompoundAssignment)]
+        public async Task TestMultiAccess()
+        {
+            await TestInRegularAndScriptAsync(
+@"public class C
+{
+    public int a;
+}
+
+public class D
+{
+    C c;
+
+    void M()
+    {
+        this.c.a [||]= this.c.a + 10;
+    }
+}",
+@"public class C
+{
+    public int a;
+}
+
+public class D
+{
+    C c;
+
+    void M()
+    {
+        this.c.a += 10;
+    }
+}");
+        }
+
+        [Fact, Trait(Traits.Feature, Traits.Features.CodeActionsUseCompoundAssignment)]
+        public async Task TestNotOnProp()
```

Comment:
Why not?

Reply:
Because it may change semantics.  There is no guarantee the prop is side-effect free.  So calling it twice may lead to a problem.  Note: this should only really be an issue if the prop is a sub-expression of the main expression.  i.e. `this.prop.x = thix.prop.x + 1`.  But i just disallowed all props for now. This could be relaxed in the future (or we could do it as part of this PR if it was felt ot be important enough).

Reply:
I don't understand. Is it not the case that there's 1 call to the setter and 1 call to the getter in each case?

Reply:
I see what you mean about `this.prop.x`, but I feel like this should be handled. It's very similar to intermediate method calls: `this.method().x`

Reply:
Alright.  i've updated to handle it where the prop is the topmost expression symbol.

Reply:
Thanks, that is awesome 🎉